src/components/Shared/ThreeBoxThreads.tsx

```diff
@@ -0,0 +1,233 @@
+    } finally {
+      if (joining) {
+        this.setState({ joiningThread: false });
+      }
+    }
+  }
+
+  private handleCommentInput = (markdown: string): void => {
+    this.commentInput = markdown;
+    this.setState({ hasCommentInput: !!this.commentInput.length });
+  }
+
+  private submitPost = async () => {
+    if (this.commentInput?.trim()) {
+      if (!await enableWalletProvider({ showNotification: this.props.showNotification })) {
+        return;
+      } else {
+        await waitUntilTrue(() => !!this.props.currentAccountAddress && getAccountIsEnabled(), 60000);
+        await this.props.threeboxLogin(this.props.currentAccountAddress);
+        await waitUntilTrue(() => !!this.props.profiles.threeBoxSpace, 60000);
+        addPost(this.state.joinedThread, this.commentInput.trim());
+      }
+    }
+  }
+
+  private deleteComment = (post: I3BoxThreadPost) => {
+    this.state.joinedThread.deletePost(post.postId);
+  }
+
+  private deleteCommentConfirmation = (post: I3BoxThreadPost): void => {
+    showSimpleMessage(
+      {
+        body: <>{i18next.t("Delete3BoxPostConfirmation")}</>,
+        buttonSpec: EnumButtonSpec.YesNo,
+        submitHandler: () => this.deleteComment(post),
+        title: i18next.t("Delete3BoxPost"),
+        width: "280px",
+      }
+    );
+  }
+
+  private handleNewPosts = async () => {
+    const posts = await this.getPosts(this.state.joinedThread);
+    this.setState({ threeboxPosts: posts });
+  }
+
+  public async componentDidMount(): Promise<void> {
+    const thread = await this.get3BoxThread();
+    if (thread) {
+      // subcribeToThread(thread, this.handleNewPosts);
+      const posts = await this.getPosts(thread);
+      this.setState({ threeboxPosts: posts });
+    } else {
+      // sp at least we know we're done setting up
+      this.setState({ threeboxPosts: [] });
+    }
+  }
+
+  static getDerivedStateFromProps(props: IProps, state: IStateProps): Partial<IStateProps> {
+    /**
+     * handle logouts
+     */
+    return (state.joinedThread && !props.profiles.threeBox) ?
+      { joinedThread: null, hasCommentInput: false, joiningThread: false } : null;
+  }
+
+  public render(): JSX.Element {
+    const daoState = this.props.daoState;
+
+    return (
+      <div className={css.container}>
+        <div className={css.startDiscussionButton}>
+          {!this.state.joinedThread && (this.state.threeboxPosts !== null) ?
+            <Tooltip placement="top" trigger={["hover"]} overlay={i18next.t(this.state.threeboxPosts?.length ? "Enable3BoxInteractions" : "CreateFirst3BoxPost")}>
+              <a onClick={this.startDiscussion}>{i18next.t(this.state.threeboxPosts?.length ? "JoinTheConversation" : "StartAConversation")}{this.state.joiningThread ? <img className={css.loading} src="/assets/images/Icon/buttonLoadingWhite.gif" /> : ""}</a>
+            </Tooltip>
+            : ""
```

Comment:
Can you please not use the pattern of empty quotation marks? Instead you can use `&&`, for example: 
`{ this.state.joinedThread && <div>your content</div> } `

I find this empty quotation marks pattern confusing and I'm trying to eliminate this across Alchemy.
I would use the ternary operation only when I have something to show if the other condition met, otherwise `&&` is much more cleaner.

Please change on the other places also.

Reply:
frankly I find your logic much more confusing because I have to understand how the boolean expression is evaluated left to right coupled with the fact that the boolean value of the <div> is totally irrelevant and that the final result of the whole expression is never actually evaluated, suggesting to me that some linters might complain and compilers/optimizers might motivated to remove the entire block.

Reply:
Further, asking me to change this logic risks creating bugs forcing me to test everything all over again.  Would have been better to asked for this sort of thing in advance.

Reply:
On further reflection I realize that contrary to what I asserted above, the final result of the whole expression actually *is* evaluated.  But it that doesn't much alter the fact that the expression has taken me a while to fully grok, a lot longer than the terniary, so I'm still not very compelled to agree that this is a simpler way of doing this.

Reply:
Maybe I'm just dense.  Bottom line is I'd prefer not have to retest all this code for this late request.

Reply:
BTW, react sucks

Reply:
If you beg me I'll do it

Reply:
OK OK I did it in the one file, but I'll go on record as saying that for me the new logic is less intuitive/clear than terniaries.

And that react sucks in any case.